Objects/unicodeobject.c

```diff
@@ -3996,6 +3996,38 @@ PyUnicode_FSDecoder(PyObject* arg, void* addr)
 }
 
 
+int
+PyUnicode_GetUTF8Buffer(PyObject *unicode, const char *errors,
+                        Py_buffer *view)
+{
+    if (!PyUnicode_Check(unicode)) {
+        PyErr_BadArgument();
+        return -1;
+    }
+    if (PyUnicode_READY(unicode) == -1) {
+        return -1;
+    }
+
+    if (PyUnicode_UTF8(unicode) != NULL) {
+        return PyBuffer_FillInfo(view, unicode,
+                PyUnicode_UTF8(unicode),
+                PyUnicode_UTF8_LENGTH(unicode),
+                1, PyBUF_SIMPLE);
+    }
+
+    PyObject *bytes = _PyUnicode_AsUTF8String(unicode, errors);
```

Comment:
Would you mind to add a comment to explain that you don't fill the internal UTF-8 cache on purpose? We have so many ways to encode Unicode to UTF-8, I never recall which one fills the cache or not.